docs/development/current_measurement.md

```diff
@@ -1,9 +1,177 @@
+
+$$V_{out} = Gain \cdot V_{s} + V_{ref}$$
+
+<figure>
+<center>
+    <img src="./images/shunt_amplifier_bidirectional.svg" height="auto" width="auto" />
+    <figcaption><b>Figure 3.</b> Shunt with bi-directional current sense amplifier.</figcaption>
+</center>
+</figure>
+
+### 2. Hall effect sensor
+
+Hall effect current measurement is a non-contact method based on the principle that for a given current flow, a proportional magnetic field is produced around the current-carrying conductor. In a [Hall sensor](https://en.wikipedia.org/wiki/Hall_effect_sensor), the current carrying conductor passes through a magnetically permeable core. The Hall effect device is mounted perpendicular to the magnetic field generated by current-carrying conductor. This produces a potential difference (voltage) that can be measured.
+
+When a hall sensor (component with X mark) like [ACS712](https://www.sparkfun.com/datasheets/BreakoutBoards/0712.pdf) is connected to the branch carrying the current of interest, due to the above principle, a voltage difference is generated at the two output terminals of the sensor. The generated voltage potential is linearly proportional to the current passing through the hall sensor. Depending on the requirement, if the generated voltage potential is found to be small, an amplifier of appropriate gain can be used to magnify the magnitude as shown in the Fig.4. This method provides isolation from the main circuit unlike shunt based method. In addition to that, it also decreases the heat dissipation. Since sensitive components are used, implementation of hall sensor based current measurement is more expensive than the former method.
+<figure>
+<center>
+    <img src="./images/hall_current_measurement.svg" height="auto" width="auto" />
+    <figcaption><b>Figure 4.</b> Principle of hall sensor based current measurement.</figcaption>
+</center>
+</figure>
+
+### 3. Inductive sensor
+
+The [inductive sensor](https://en.wikipedia.org/wiki/Inductive_sensor) consists of a wire-wound core through which current carrying conductor passess. AC current flowing through this conductor constantly changes potential from positive to negative and back again. This results in expanding and collapsing magnetic field which induces voltage in the windings. Further inductive sensor converts this induced voltage into proportional voltage which will be given as output. Hence, in this case, voltage is the measure of change in magnetic field strength and not direct measure of magnetic field as in the case of Hall effect sensor. That's why inductive sensors are only suitable for measuring AC currents over other methods. Like Hall sensor, this also provides excellent isolation from the main circuit and has the advantage of very low noise when compared with the former. 
+
+#### Conclusion
```

Comment:
Should be ### as this should be on the same indentation level as the previous three sections.